editoast/fga/src/lib.rs

```diff
@@ -446,10 +446,7 @@ macro_rules! fga {
 
     // fga!(Group:"my_friends"#member) => userset syntax "group:my_friends#member"
     ($ty:ident : $id:literal # $relation:ident) => {
-        {
-            use $crate::model::Relation as _;
-            $ty::$relation().userset(&fga!($ty:$id))
-        }
+        $ty::$relation().userset(&fga!($ty:$id))
```

Comment:
This change seems weird. `.userset()` is a function of `Relation`'s trait, so adding the `use` inside the macro seems a security for the macro to always work (in file that already import that traits but also in traits that don't import that trait).

Reply:
Having a block section here makes the borrow checker fail. I will try another way to fix it, keeping the `use`. 
The main issue here is using recursive use of the macro.

Reply:
It's not serious enough to block the 2024 edition, but keeping the macro hygienic is important. If the recursion is the problem, you can totally inline the output of the branch `$ty:type : $id:ident`

Reply:
Or inline the trait with something like this, I believe this should work too (not tested)

```rust
$crate::model::Relation::userset($ty::$relation(), &fga!($ty:$id))
```

Reply:
Fixed